extensions-core/datasketches/src/main/java/org/apache/druid/query/aggregation/datasketches/hll/HllSketchBuildAggregatorFactory.java

```diff
@@ -221,11 +226,21 @@ private HllSketchUpdater formulateSketchUpdater(ColumnSelectorFactory columnSele
             }
           };
           break;
+        case ARRAY:
+          final ExpressionType expressionType = ExpressionType.fromColumnTypeStrict(capabilities);
+          updater = sketch -> {
+            final Object o = selector.getObject();
+            if (o != null) {
+              byte[] bytes = ExprEval.toBytes(expressionType, o);
+              sketch.get().update(bytes);
+            }
+          };
+          break;
```

Comment:
My intention was for `processAsArrays` to only have effect when the input types are unknown, e.g. ingestion time, to serve as a switch on whether or not to treat `List` as a MVD or an ARRAY. If we have the actual type information it means we are operating on top of a column, and so my plan was that this setting has no effect and it should definitely use array processing. 

I was not imagining `processAsArray` as a feature flag on whether or not we handle arrays at all

Reply:
But didn't you say that there could be places even during ingestion that an `ARRAY` might show up as a `java.lang.List`?  If we are adding the flag, it seems easier to reason about and think consistently if it applies equivalent across arrays and lists and we make the question of what happens at ingest time a question of how the person writes the ingestion spec.

I.e. said another way, I think it's easier to document the parameter as a flag that applies consistently to array-like things.  Than it is to document it as a parameter that applies to objects which end up as `java.lang.List` but not to objecst that end up as an `Object[]` and then subsequently explain why and when we expect an `Object[]` instead of a `java.lang.List` and how someone can affect which object they are getting.

Reply:
its not necessary to discuss since i have removed these changes from this PR, but this case statement was a clear indicator that an input is *definitely* an array. It should not process it in any other way, so for now i just throw an invalid input exception if this is the case, instead of incorrectly/silently falling into some other case to process stuff as an MVD.